metamorph/store/dynamodb/dynamodb.go

```diff
@@ -0,0 +1,461 @@
+package dynamodb
+
+import (
+	"context"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	"github.com/bitcoin-sv/arc/metamorph/metamorph_api"
+	"github.com/bitcoin-sv/arc/metamorph/store"
+	"github.com/libsv/go-p2p/chaincfg/chainhash"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	"github.com/opentracing/opentracing-go/log"
+	"github.com/ordishs/gocore"
+)
+
+type DynamoDB struct {
+	dynamoCli *dynamodb.Client
+}
+
+const ISO8601 = "2006-01-02T15:04:05.999Z"
+
+func New() (store.MetamorphStore, error) {
+	// Using the SDK's default configuration, loading additional config
+	// and credentials values from the environment variables, shared
+	// credentials, and shared configuration files
+	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion(os.Getenv("AWS_REGION")))
+	if err != nil {
+		return &DynamoDB{}, err
+	}
+
+	// create dynamodb client
+	dynamodbClient := &DynamoDB{dynamoCli: dynamodb.NewFromConfig(cfg)}
+
+	// create table if not exists
+	exists, err := dynamodbClient.TableExists("transactions")
+	if err != nil {
+		return dynamodbClient, err
+	} else if !exists {
+		err = dynamodbClient.CreateTransactionsTable()
+		if err != nil {
+			return dynamodbClient, err
+		}
+	}
+
+	// create table if not exists
+	exists, err = dynamodbClient.TableExists("blocks")
+	if err != nil {
+		return dynamodbClient, err
+	} else if !exists {
+		err = dynamodbClient.CreateBlocksTable()
+		if err != nil {
+			return dynamodbClient, err
+		}
+	}
+
+	// Using the Config value, create the DynamoDB client
+	return dynamodbClient, nil
+}
+
+func (ddb *DynamoDB) TableExists(tableName string) (bool, error) {
+	// Build the request with its input parameters
+	resp, err := ddb.dynamoCli.ListTables(context.TODO(), &dynamodb.ListTablesInput{})
```

Comment:
Alternatively here instead of listing the table could be described in order to check if it exists like this
```
resp, err := ddb.dynamoCli.DescribeTable(context.TODO(), &dynamodb.DescribeTableInput{TableName: &tableName})
```

Then it wouldn't be needed to iterate through them.
But this is also fine